lib/generate-id.js

```diff
@@ -21,12 +20,50 @@ const findId = async ( model, length ) => {
   return id
 }
 
+const idTypes = {
+  shabad: [ Shabads, 3 ],
+  line: [ Lines, 4 ],
+}
+
 /**
- * Generates a line id and a shabad id, using the database.
- */
+  * Logs the id to the console
+  * @param {Number} count the count of id to generate
+  * @param {String} type check `availableTypes`
+  */
+const logGenerateIds = async ( count, type ) => {
+  const idType = idTypes[ type ]
+  const [ Model, idLength ] = idType
+
+  // Grab ids to exclude from database
+  const ids = new Set( ( await Model.query().select( 'id' ) ).map( ( { id } ) => id ) )
+
+  // Loop to print id's and adds the generated id in the Set
+  for ( let i = 0; i < count; i += 1 ) {
+    const newId = await findId( ids, idLength )
+    ids.add( newId )
+    console.log( newId )
+  }
+}
+
+/**
+  * Interpret user command and print the id, using the database.
+  */
 const main = async () => {
-  console.log( 'New Line ID: '.bold.cyan + `${await findId( Lines, 4 )}`.rainbow )
-  console.log( 'New Shabad ID: '.bold.magenta + `${await findId( Shabads, 3 )}`.rainbow )
+  commander
+    .version( '0.0.1' )
+    .description( "Generate ID\'s for ShabadOS Database \n\n Type\n shabad \t\tgenerate id\'s for shabads \nline \t\tgenerate id\'s for line" )
+    .option( '-c, --count <integer>', 'number of ID\'s', 1 )
+    .requiredOption( '-t, --type [type]', 'generate ID for only [type]' )
+    .parse( process.argv )
+
+  const { count, type } = commander
+  try {
+    await logGenerateIds( count, type )
+  } catch ( e ) {
```

Comment:
What if there is another type of error?

Reply:
Currently, this will suppress all other errors mistakenly.

https://github.com/tj/commander.js#custom-option-processing on how to supply a 4th parameter which is a custom validator/processor, throw the error there, and then `console.error(err.message)` once here

Reply:
so if it is something other than invalid type error then it should throw the `console.error(err.message)` otherwise do what I already have there? right?

Reply:
> What if there is another type of error?

I can have a conditional catch-block

```
try {
    await logGenerateIds( count, type )
  } catch ( e ) {
    if ( e instanceof TypeError ) {
      console.log( 'Invalid value for type parameter\n' )
      console.log( 'Available types are: \n' )
      console.log( Object.keys( idTypes ).join( '\n' ) )
    } else {
      console.error( e )
    }
  }
```